Set initial size for AGL

jkim-julie opened the pull request:
It sets initial values for App size with AGL window size. It
couldn't get the AGL window size for now as Window manager doesn't
support that functionality for now. So, it sets the default values
for m_widthOverride and m_heightOverride as these values delivers
to WebView size and window size.

[SPEC-1890] WebApp View Size doesn't fit the homescreen.
https://jira.automotivelinux.org/browse/SPEC-1890

jkim-julie commented:
@jdapena, it fixes WebView size issue. I described the investigation for it at https://jira.automotivelinux.org/browse/SPEC-1890. I tried to set window-size as well but it didn't work. So, I added initial size value for web apps with AGL window size. I could see m_widthOverride and m_heightOverride are delivered to window size and webview size. So, I used them. If there is any suggestion to set the size properly, please let me know.

jdapena commented on src/core/ApplicationDescription.cpp:
```diff
@@ -25,6 +25,11 @@
+// The size from AGL Window.
+// https://jira.automotivelinux.org/browse/SPEC-1890
+constexpr unsigned int kWebAppWidth = 608;
+constexpr unsigned int kWebAppHeight = 837;
```
I'd prefer we'd avoid hardcoding this here. If we are not getting any resolution for platform, at least the value should be added in the AGL files.

The solution could be just provide the resolution from environment variable (i.e. WAM_DEFAULT_OVERRIDE_HEIGHT and WAM_DEFAULT_OVERRIDE_WIDTH), and use the values in the WebAppWaylandAGL to set the overrides in case app does not provide its own override.

BTW, is that resolution of 608x837 provided by the application manifest? I guess the answer is no, but...

I'd prefer that we'd add those env vars from wam.bb script, and it would be even better if we could know the values at yocto level instead of hardcoding them even in the script.

jkim-julie committed:
Set initial size for AGL

It sets initial values for App size with AGL window size. It
couldn't get the AGL window size for now as Window manager doesn't
support that functionality for now. So, it sets the default values
for m_widthOverride and m_heightOverride as these values delivers
to WebView size and window size after getting the size value from
Env variables.

It adds 'WAM_WINDOW_SIZE' to WebAppMgr.env and passes it through
WebAppManagerConfig.

[SPEC-1890] WebApp View Size doesn't fit the homescreen.
https://jira.automotivelinux.org/browse/SPEC-1890

jkim-julie commented:
@jdapena, as we discussed at the channel, the patch was updated with using Env variable at WebAppMgr.env at this repo. Could you take a look?

jdapena commented on src/core/WebAppManagerConfig.cpp:
```diff
@@ -17,6 +17,7 @@
 #include "WebAppManagerConfig.h"
 
 #include <unistd.h>
+#include <stdlib.h>
```
Please use cstdlib. This means also using std::getenv.

We are moving to C++ headers as needed.

jkim-julie replied:
Done.

jkim-julie commented:
@jdapena , I updated the branch. Please take a look.
